src/pdm/cli/commands/run.py

```diff
@@ -58,6 +59,24 @@ def replace(m: re.Match[str]) -> str:
     return interpolated, count > 0
 
 
+def _interpolate_pdm(script: str) -> tuple[str, bool]:
+    """Interpolate the `{pdm} placeholder in a string"""
+
+    def replace(m: re.Match[str]) -> str:
+        return sh_join([sys.executable, "-m", "pdm"])
+
+    interpolated, count = RE_PDM_PLACEHOLDER.subn(replace, script)
+    return interpolated, count > 0
+
+
+def interpolate(script: str, args: Sequence[str]) -> tuple[str, bool]:
+    """Interpolate the `{args:[defaults]} placeholder in a string"""
+
+    script, args_interpolated = _interpolate_args(script, args)
+    script, pdm_interpolated = _interpolate_pdm(script)
+    return script, args_interpolated or pdm_interpolated
```

Comment:
You can't mix args_interpolated with pdm_interpolated, since the former is needed by the runner to know whether it needs to append other arguments.